assets/media-gallery.js

```diff
@@ -0,0 +1,101 @@
+if (!customElements.get('media-gallery')) {
+  customElements.define('media-gallery', class MediaGallery extends HTMLElement {
```

Comment:
This is clever.  Using this approach we might even be able to inline some of our smaller custom elements.  Keeping this one in the back of my mind for asset management ideas.